Wikipedia/Code/Article as a Living Document/ArticleAsLivingDocLargeEventCollectionViewCell.swift

```diff
@@ -0,0 +1,357 @@
+        let image = UIImage(named: "thank")
+        return actionButton(with: image, text: WMFLocalizedString("aaald-events-thank-title", value: "Thank", comment: "Button title that thanks users for their edit in article as a living document screen"))
+    }()
+    private lazy var viewChangesButton: AlignedImageButton = {
+        let image = UIImage(named: "document")
+        return actionButton(with: image, text: WMFLocalizedString("aaald-view-changes", value: "View changes", comment: "Button title on a article as a living document cell that sends user to the revision history screen."))
+    }()
+    private lazy var viewDiscussionButton: AlignedImageButton = {
+        let image = UIImage(named: "document")
+        return actionButton(with: image, text: WMFLocalizedString("aaald-view-discussion", value: "View discussion", comment: "Button title on an article as a living document cell that sends a user to the event's talk page topic."))
+    }()
+    
+    let timelineView = TimelineView()
+
+    weak var delegate: ArticleAsLivingDocHorizontallyScrollingCellDelegate?
+    weak var articleDelegate: ArticleDetailsShowing?
+    
+    private var largeEvent: ArticleAsLivingDocViewModel.Event.Large?
+    private var changeDetails: [ArticleAsLivingDocViewModel.Event.Large.ChangeDetail] = []
+    
+    private var flowLayout: UICollectionViewFlowLayout? {
+        return collectionView.collectionViewLayout as? UICollectionViewFlowLayout
+    }
+    private let collectionView = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
+    
+    private var collectionViewHeight: CGFloat = 0
+    
+    override func setup() {
+        contentView.addSubview(descriptionLabel)
+        contentView.addSubview(userInfoTextView)
+        contentView.addSubview(timestampLabel)
+        contentView.addSubview(collectionView)
+        timelineView.isOpaque = true
+        timestampLabel.isOpaque = true
+        descriptionLabel.isOpaque = true
+        userInfoTextView.isOpaque = true
+
+        userInfoTextView.delegate = self
+        
+        timelineView.decoration = .singleDot
+        contentView.insertSubview(timelineView, belowSubview: collectionView)
+        
+        wmf_configureSubviewsForDynamicType()
+        
+        descriptionLabel.numberOfLines = 0
+        flowLayout?.scrollDirection = .horizontal
+        collectionView.register(ArticleAsLivingDocSnippetCollectionViewCell.self, forCellWithReuseIdentifier: ArticleAsLivingDocSnippetCollectionViewCell.identifier)
+        collectionView.register(ArticleAsLivingDocReferenceCollectionViewCell.self, forCellWithReuseIdentifier: ArticleAsLivingDocReferenceCollectionViewCell.identifier)
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.showsHorizontalScrollIndicator = false
+        collectionView.showsVerticalScrollIndicator = false
+        collectionView.alwaysBounceHorizontal = true
```

Comment:
I think I was just basing this off of how On This Day worked, but maybe worth asking @carolynlimadeo what she thinks. Since article snippet and reference cells are interspersed I do think they should be consistent, either all bouncing or none bouncing.

Reply:
Snippet cells always have just 1 item, right? But References have many? That seems like a key distinction: Never possibly going to need to scroll, it seems superfluous to have bounce. But don't feel super-strongly as we get this near the finish line.

Reply:
It's a bit of a mix, a large event cell can have any mix snippet and reference cells. For example you could have a single large event cell with 3 side scrolling cells: 1 snippet cell of added text in the article, 1 reference cell with details of a new reference, and 1 snippet cell of an updated article title description. You also might have a large event cell that only has 1 snippet cell of added text in the article, because they didn't make any other change types in that particular revision.

There is a use case of a large event cell that consistently would only ever have 1 side scrolling cell - a new talk page topic. So there's that, we might be able to remove bouncing there. We'd probably still want to keep scrolling though in case their dynamic type size is bumped up enough for the cell to go off screen, or we could add some additional logic for talk page topics to only go as wide as the view controller's view so it could never go off screen.

It's probably better that we fix these later if we decide it's a must. I think it'd make more sense to hash it out as a part of [T265672](https://phabricator.wikimedia.org/T265672) if/when we tackle that since it will touch a bunch of the side-scrolling cell sizing.

Reply:
Sounds good to me!